src/auto-complete.jsx

```diff
@@ -349,8 +355,9 @@ const AutoComplete = React.createClass({
     };
 
     const requestsList = [];
+    const maxSearchResult = this.props.maxSearchResult;
 
-    this.props.dataSource.map((item) => {
+    for (let item of this.props.dataSource) {
```

Comment:
I want to break the loop on some condition, but forEach don't have `break` or `return false` implementation.

Then we have `every`, `some` or `for` loop, I thought `for` loop gives better understandability  so used it